### PR TITLE
Added a configuration flag enabling to tail all container logs

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Added default source configuration forcing all containers to be tailed when config flag `logs_config.container_collect_all` or environment variable `DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL` is set to true.

### Motivation

Ease the configuration of logs-agent when running in container.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -213,6 +213,7 @@ func init() {
 	BindEnvAndSetDefault("logs_config.dev_mode_use_proto", false)
 	BindEnvAndSetDefault("logs_config.run_path", defaultRunPath)
 	BindEnvAndSetDefault("logs_config.open_files_limit", 100)
+	BindEnvAndSetDefault("logs_config.collect_all", false)
```
Comment (tmichelet) [inline pkg/config/config.go]: `collect_all` seems very vague, @NBParis how would you name this? `collect_all_docker_logs`?

**Review by tmichelet — DISMISSED** [review]

File: pkg/config/config.go (hunk 1 above)
Comment (NBParis) [reply pkg/config/config.go]: @tmichelet Indeed it is clearer but it is so long.
What do you think about `logs_config.docker_collect_all` ?

Comment (tmichelet) [reply pkg/config/config.go]: ok

Comment (ajacquemot) [reply pkg/config/config.go]: I'd vote for `logs_config.container_collect_all`

**Review by tmichelet — DISMISSED** [review]

File: releasenotes/notes/logs_docker_environment_variable-daf65e8291a6d309.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Added an environment variable `DD_LOGS_CONFIG_COLLECT_ALL` to enable logs tailing on all containers.
```
Comment (olivielpeau) [inline releasenotes/notes/logs_docker_environment_variable-daf65e8291a6d309.yaml]: you haven't updated the release note with the updated config option name ;)

Comment (ajacquemot) [reply releasenotes/notes/logs_docker_environment_variable-daf65e8291a6d309.yaml]: Good catch, thanks

**Review by olivielpeau — APPROVED** [review]

**Review by tmichelet — APPROVED** [review]

**Pull request merged**